gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -231,27 +232,32 @@ def run(argv=None):
   processed_variant_factory = processed_variant.ProcessedVariantFactory(
       header_fields,
       known_args.split_alternate_allele_info_fields,
-      known_args.annotation_fields)
+      known_args.annotation_fields,
+      beam_util.CounterFactory())
 
   pipeline_options = PipelineOptions(pipeline_args)
-  with beam.Pipeline(options=pipeline_options) as p:
-    variants = _read_variants(p, known_args)
-    variants |= 'FilterVariants' >> filter_variants.FilterVariants(
-        reference_names=known_args.reference_names)
-    if variant_merger:
-      variants |= (
-          'MergeVariants' >> merge_variants.MergeVariants(variant_merger))
-    proc_variants = variants | 'ProcessVaraints' >> beam.Map(
-        processed_variant_factory.create_processed_variant).\
-      with_output_types(processed_variant.ProcessedVariant)
-    _ = (proc_variants |
-         'VariantToBigQuery' >> variant_to_bigquery.VariantToBigQuery(
-             known_args.output_table,
-             header_fields,
-             variant_merger,
-             processed_variant_factory,
-             append=known_args.append,
-             omit_empty_sample_calls=known_args.omit_empty_sample_calls))
+  pipeline = beam.Pipeline(options=pipeline_options)
+  variants = _read_variants(pipeline, known_args)
+  variants |= 'FilterVariants' >> filter_variants.FilterVariants(
+      reference_names=known_args.reference_names)
+  if variant_merger:
+    variants |= (
+        'MergeVariants' >> merge_variants.MergeVariants(variant_merger))
+  proc_variants = variants | 'ProcessVaraints' >> beam.Map(
+      processed_variant_factory.create_processed_variant).\
+    with_output_types(processed_variant.ProcessedVariant)
+  _ = (proc_variants |
+       'VariantToBigQuery' >> variant_to_bigquery.VariantToBigQuery(
+           known_args.output_table,
+           header_fields,
+           variant_merger,
+           processed_variant_factory,
+           append=known_args.append,
+           omit_empty_sample_calls=known_args.omit_empty_sample_calls))
+  result = pipeline.run()
+  result.wait_until_finish()
+
+  beam_util.log_all_counters(result)
```

Comment:
For my information: since it was the first use of beam counter in this code base (right?), I am wondering what were the motivations behind it? :)

Reply:
Right this is the first use and I have tried to make subsequent uses simple by adding the beam_util library.

Part of the motivation is what I mentioned in my first comments after sending the PR (I am not sure if GitHub sent you emails for that, if not please take a look at them in the UI, in particular the snapshot I have attached from Dataflow dashboard). There are at least two benefits:

1) In error/warning cases, (e.g., `annotation_alt_mismatch_counter`) it is hard to find/monitor those through logs since each worker has its own logs (I am not even sure if the Dataflow/Stackdriver dashboards easily expose those logs but certainly you could get them in the Borg world). One higher level way is to count those cases and log them at the end. So at least one can easily/quickly check if their job ran error/warning free.

2) Non-error counters can be used for monitoring the progress of the job. For example, I vaguely know that there are N variants in my input and through Dataflow dashboard pipeline I can see live how many of those records have been processed. It is needless to say that monitoring error counters is also very useful: If I get too many errors while the pipeline is running I can stop it early and investigate the issue instead of waiting until it finished and potentially produce garbage results.